### PR TITLE
Added filters before updates to submit only changed votes

### PR DESCRIPTION
**Pull request by tommylau523**

### CODE REVIEW

**Comment by tommylau523** [comment]
* **#2383** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2383?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tommylau523 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/scenes/AcceleratorRouter/Voting/VotingEdit.tsx [hunk 1]
```diff
@@ -47,11 +48,17 @@ const VotingEdit = () => {
     if (votes.some((vote) => vote.voteOption === 'Conditional' && !vote.conditionalInfo)) {
       return;
     }
+
+    const updatedVotes = votes.filter(({ conditionalInfo, userId, voteOption }) => {
+      const orignal = phaseVotes.votes.find((originalVote) => originalVote.userId === userId)!!;
+      return orignal.conditionalInfo === conditionalInfo && orignal.voteOption === voteOption;
```
Comment (karthikbtf) [inline src/scenes/AcceleratorRouter/Voting/VotingEdit.tsx:54]: should this be `original.conditionalInfo !== conditionalInfo || original.voteOption !== voteOption`?

File: src/scenes/AcceleratorRouter/Voting/UserVote.tsx [hunk 2]
```diff
@@ -131,15 +137,15 @@ export const UserVoteEdit = ({
  * label showing Voter: <name>
  */
 type UserLabelProps = {
-  vote: VoteSelection;
+  user?: UserIdentity;
```
Comment (karthikbtf) [inline src/scenes/AcceleratorRouter/Voting/UserVote.tsx:140]: Nice cleanup, thanks 🙏

File: src/scenes/AcceleratorRouter/Voting/VotingEdit.tsx (hunk 1 above)
Comment (tommylau523) [reply src/scenes/AcceleratorRouter/Voting/VotingEdit.tsx:54]: You are so right I forgot to negate the whole thing

**Review by karthikbtf — APPROVED** [review]
Nice 👍

**Comment by tommylau523** [comment]
### Merge activity

* **Mar 21, 5:31 PM EDT**: @tommylau523 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2383).

**Pull request merged**